From: Eike Hein <hein@kde.org>
Date: Tue, 22 Dec 2015 18:26:15 +0000
Subject: Stop redirecting windows when item is disabled or hidden.
X-Git-Url: http://quickgit.kde.org/?p=plasma-framework.git&a=commitdiff&h=c64a94a265acd5003fbbb4e3abc7fdb72726b4c3
---
Stop redirecting windows when item is disabled or hidden.

Concretely fixes Task Manager tooltips slowing down app rendering even
after the tooltip is hidden.

REVIEW:126475
BUG:356938
---


--- a/src/declarativeimports/core/windowthumbnail.cpp
+++ b/src/declarativeimports/core/windowthumbnail.cpp
@@ -98,6 +98,20 @@
         stopRedirecting();
         startRedirecting();
     });
+    connect(this, &QQuickItem::enabledChanged, [this]() {
+        if (!isEnabled()) {
+            stopRedirecting();
+        } else {
+            startRedirecting();
+        }
+    });
+    connect(this, &QQuickItem::visibleChanged, [this]() {
+        if (!isVisible()) {
+            stopRedirecting();
+        } else {
+            startRedirecting();
+        }
+    });
     if (QGuiApplication *gui = dynamic_cast<QGuiApplication *>(QCoreApplication::instance())) {
         m_xcb = (gui->platformName() == QStringLiteral("xcb"));
         if (m_xcb) {
@@ -148,7 +162,11 @@
     }
     stopRedirecting();
     m_winId = winId;
-    startRedirecting();
+
+    if (isEnabled() && isVisible()) {
+        startRedirecting();
+    }
+
     emit winIdChanged();
 }
 


